tokeniser.go

```diff
@@ -17,38 +18,60 @@ const (
 
 	// double represents "double quoted" parts of a BAPS3 message.
 	double
+
+	// bufsize is the number of bytes the Tokeniser will try to read from
+	// its Reader in one go.
+	bufsize int = 4096
 )
 
-// Tokeniser holds the state of a BAPS3 protocol tokeniser.
+// Tokeniser holds the state of a Bifrost protocol tokeniser.
 type Tokeniser struct {
+	// raw is the current back-buffer of bytes to tokenise.
+	raw []byte
+	// rawpos is the current position in `raw`.
+	rawpos int
+	// rawcount is the number of valid bytes in `raw`.
+	rawcount int
+
 	inWord           bool
 	escapeNextChar   bool
 	currentQuoteType quoteType
 	word             *bytes.Buffer
 	words            []string
```

Comment:
I wonder if it would be better if there was a clearer distinction between whether we are reading strings or byte slices